cmd/puppeth/wizard_node.go

```diff
@@ -77,6 +77,35 @@ func (w *wizard) deployNode(boot bool) {
 			infos.ethashdir = w.readDefaultString(infos.ethashdir)
 		}
 	}
+	// Potentially use custom version of geth
+	fmt.Println()
+	fmt.Println("Which geth should be used? (default = latest)")
+	fmt.Println(" 1. Use latest release")
+	fmt.Println(" 2. Specify git repository")
+	choice := w.read()
+	switch {
+	case choice == "" || choice == "1":
+		break
+	case choice == "2":
+		fmt.Println()
+		fmt.Println("Where's the git repository? (http/https url)")
+		url := w.readURL()
+
+		switch url.Scheme {
+		case "http", "https":
+			infos.gitRepo = url.String()
+			fmt.Println()
+			fmt.Println("What branch? (default = repository default)")
```

Comment:
You specify that `empty` will be the default branch, but your clone command always specified the branch `--branch {{.GitBranch}}`. As such, if none is given, the clone will fail

```
$ git clone https://github.com/ethereum/go-ethereum --branch --single-branch
Cloning into 'go-ethereum'...
fatal: Remote branch --single-branch not found in upstream origin
```